### ISSUE TITLE
security vulnerabilities upon npm install

### ISSUE THREAD

**Issue by mroswell** [h-enk/doks#340]
## Description

```
% npm install
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.

added 1247 packages, and audited 1248 packages in 24s

98 packages are looking for funding
  run `npm fund` for details

19 vulnerabilities (11 moderate, 8 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
npm notice
npm notice New minor version of npm available! 7.7.6 -> 7.17.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v7.17.0
npm notice Run npm install -g npm@7.17.0 to update!
npm notice
```

After running `npm audit fix` I get 

```
added 2 packages, removed 1 package, changed 17 packages, and audited 1249 packages in 12s

99 packages are looking for funding
  run `npm fund` for details

# npm audit report

glob-parent  <5.1.2
Severity: moderate
Regular expression denial of service - https://npmjs.com/advisories/1751
fix available via `npm audit fix`
node_modules/glob-parent
  chokidar  1.0.0-rc1 - 2.1.8
  Depends on vulnerable versions of glob-parent
  node_modules/watchpack-chokidar2/node_modules/chokidar
    watchpack-chokidar2  *
    Depends on vulnerable versions of chokidar
    node_modules/watchpack-chokidar2
      watchpack  1.7.2 - 1.7.5
      Depends on vulnerable versions of watchpack-chokidar2
      node_modules/watchpack
        webpack  4.44.0 - 4.46.0
        Depends on vulnerable versions of watchpack
        node_modules/webpack

normalize-url  <=4.5.0 || 5.0.0 - 5.3.0 || 6.0.0
Severity: high
Regular Expression Denial of Service - https://npmjs.com/advisories/1755
fix available via `npm audit fix --force`
Will install hugo-bin@0.30.0, which is a breaking change
node_modules/normalize-url
  cacheable-request  0.1.0 - 6.0.0
  Depends on vulnerable versions of normalize-url
  node_modules/cacheable-request
    got  8.0.0 - 9.5.0
    Depends on vulnerable versions of cacheable-request
    node_modules/got
      download  >=7.0.0
      Depends on vulnerable versions of got
      node_modules/download
        bin-wrapper  2.1.2 || >=4.0.0
        Depends on vulnerable versions of download
        node_modules/bin-wrapper
          hugo-bin  >=0.31.0
          Depends on vulnerable versions of bin-wrapper
          node_modules/hugo-bin

trim-newlines  <3.0.1 || =4.0.0
Severity: high
Regular Expression Denial of Service - https://npmjs.com/advisories/1753
fix available via `npm audit fix`
node_modules/get-pkg-repo/node_modules/trim-newlines
  meow  3.4.0 - 5.0.0
  Depends on vulnerable versions of trim-newlines
  node_modules/get-pkg-repo/node_modules/meow

13 vulnerabilities (5 moderate, 8 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force
```

### Steps to reproduce

% npm install

### Expected result

0 vulnerabilities
### Actual result

19 or 13 vulnerabilities, including 8 high
### Environment

```
> doks-child-theme@0.2.3 precheck
> npm version

{
  'doks-child-theme': '0.2.3',
  npm: '7.7.6',
  node: '15.14.0',
  v8: '8.6.395.17-node.28',
  uv: '1.41.0',
  zlib: '1.2.11',
  brotli: '1.0.9',
  ares: '1.17.1',
  modules: '88',
  nghttp2: '1.42.0',
  napi: '8',
  llhttp: '2.1.3',
  openssl: '1.1.1k+quic',
  cldr: '38.1',
  icu: '68.2',
  tz: '2020d',
  unicode: '13.0',
  ngtcp2: '0.1.0-DEV',
  nghttp3: '0.1.0-DEV'
}

> doks-child-theme@0.2.3 check
> hugo version

hugo v0.82.0-9D960784+extended darwin/amd64 BuildDate=2021-03-21T17:28:04Z VendorInfo=gohugoio
```

**Comment by mroswell** [comment]
I'll add this for more complete info:
% npm outdated
Package                    Current  Wanted  Latest  Location                                Depended by
flexsearch                  0.6.32  0.6.32   0.7.1  node_modules/flexsearch                 hugonewbie
highlight.js                10.7.3  10.7.3  11.0.1  node_modules/highlight.js               hugonewbie
hugo-bin                    0.70.2  0.70.2  0.71.1  node_modules/hugo-bin                   hugonewbie
stylelint-config-standard   21.0.0  21.0.0  22.0.0  node_modules/stylelint-config-standard  hugonewbie

**Comment by h-enk** [comment]
Hi @mroswell, thanks for reporting this. The security issues lie upstream w/ `hugo-bin` (I think of replacing this one), `standard-version` (not needed anyway, I will remove this one), and `netlify-lambda` (optional tool, I think of dropping this one). They are aware and will update their packages — soon. 

For now — if you have Hugo globally installed and do not use Netlify Lambda — the following steps will work:

1. `git clone https://github.com/h-enk/doks-child-theme.git my-doks-site`
2. In `package.json`, delete `devDependencies`:
    - `"hugo-bin": "^0.70",`
    - `"netlify-lambda": "^2.0",`
    - `"standard-version": "^9.1",`
3. Run `npm install`
4. Run `npm update`

Result: `found 0 vulnerabilities`

**Comment by h-enk** [comment]
Addressed upstream: https://github.com/fenneclab/hugo-bin/issues/114, https://github.com/netlify/netlify-lambda/issues/456, and https://github.com/conventional-changelog/standard-version/issues/772,

**Comment by mroswell** [comment]
Thanks for sharing this. 

What action do I take in my project to perform the update?

A note, when I run: 
```
% npm outdated
```
I get a list of packages that only shares one item in common with the list of three above.
```
Package                    Current  Wanted  Latest  Location                                Depended by
eslint                      7.28.0  7.29.0  7.29.0  node_modules/eslint                     hugonewbie
flexsearch                  0.6.32  0.6.32   0.7.1  node_modules/flexsearch                 hugonewbie
highlight.js                10.7.3  11.0.1  11.0.1  node_modules/highlight.js               hugonewbie
hugo-bin                    0.70.2  0.71.1  0.71.1  node_modules/hugo-bin                   hugonewbie
stylelint-config-standard   21.0.0  22.0.0  22.0.0  node_modules/stylelint-config-standard  hugonewbie
```

**Comment by h-enk** [comment]
Fixed w/ 93f7598f872509805287b24d7d06fa3e6f745fbf

- Replaced  [hugo-bin](https://github.com/fenneclab/hugo-bin) w/ [hugo-installer](https://github.com/dominique-mueller/hugo-installer)
- Replaced [standard-version](https://github.com/conventional-changelog/standard-version) w/ [auto-changelog](https://github.com/cookpete/auto-changelog)
- Replaced [netlify-lambda](https://github.com/netlify/netlify-lambda) w/ [Netlify Functions bundler](https://www.netlify.com/blog/2021/04/02/modern-faster-netlify-functions/)
